src/cim/components/Agent/AgentsSelectionTable.tsx

```diff
@@ -22,21 +22,38 @@ import DefaultEmptyState from '../../../common/components/ui/uiState/EmptyState'
 import { usePagination } from '../../../common/components/hosts/usePagination';
 import { useFormikHelpers } from '../../../common/hooks/useFormikHelpers';
 import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
+import { HostDetail } from '../../../common/components/hosts/HostRowDetail';
+import { ExpandComponentProps } from '../../../common/components/hosts/AITable';
+import { Host } from '../../../common/api/types';
+import { onDiskRoleType } from '../../../common/components/hosts/DiskRole';
+import { HostsTableActions } from '../../../common/components/hosts/types';
 
 type AgentsSelectionTableProps = {
   matchingAgents: AgentK8sResource[];
   width?: number;
   onEditRole?: AgentTableActions['onEditRole'];
+  onSetInstallationDiskId?: AgentTableActions['onSetInstallationDiskId'];
   onEditHost?: AgentTableActions['onEditHost'];
   onHostSelect?: VoidFunction;
 };
 
+export const getExpandComponent = (
```

Comment:
this is anti-pattern as every call of `getExpandComponent` will create a new component which breaks React's reconciliation cycle. To pass the additional props to `HostDetail` we can use context. This is the same as https://github.com/openshift-assisted/assisted-ui-lib/pull/1528#discussion_r927518445

Reply:
Done